llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp

```diff
@@ -3214,7 +3223,9 @@ bool X86AsmParser::ParseInstruction(ParseInstructionInfo &Info, StringRef Name,
       else if (Prefix == "vex3")
         ForcedOpcodePrefix = OpcodePrefix_VEX3;
       else if (Prefix == "evex")
-        ForcedOpcodePrefix = OpcodePrefix_EVEX;
+        ForcedOpcodePrefix = is64BitMode() && ifCmpOrTestOpcode(Opcode)
```

Comment:
`OpcodePrefix_EVEX_CMP_TEST` is introduced mainly for here. If not introduced, `ForcedOpcodePrefix` will be assigned `OpcodePrefix_EVEX`. And ...

Reply:
We can check `isCMP` `isTEST` around line  4210 to resolve this.

```
if ((TSFlags & X86II::EncodingMask) != X86II::EVEX)
```

Reply:
Will try

Reply:
2dd7c4a